x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java

```diff
@@ -477,11 +503,32 @@ public void compareAndExchangeRegister(
                 }
             }
 
-            listener.onResponse(
-                OptionalBytesReference.of(
-                    registers.computeIfAbsent(key, ignored -> new BytesRegister()).compareAndExchange(expected, updated)
-                )
-            );
+            final BytesReference witness;
+            synchronized (registerMutex) {
+                witness = registers.computeIfAbsent(key, ignored -> new BytesRegister()).compareAndExchange(expected, updated);
```

Comment:
Does this synchronization introduce an artificial serialization of operations that means we are not really testing `compareAndExchange` in a concurrent fashion? Is this a problem?

Reply:
No, the correctness of this `compareAndExchange` (itself implemented as a `synchronized` method) isn't really the subject of this test. I added a comment in 2b4b3c737fb8aa93f09332c774e8974655909cc7.